### PR TITLE
feat: creating users as juris admin

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/2763)

- [x] This change addresses the issue in full

## Description
This handles the case of creating users as a jurisdictional admin

## How Can This Be Tested/Reviewed?
log in as a jurisdictional admin

navigate to the user grid, and move to create a user

first create another jurisdictional admin, notice that the fields match up with the mock and that the save happens successfully (the new juris admin has the same jurisdiction as the creating jurisdictional admin)

move to create a new parter, notice that the fields match up with the mock, and they are only allowed to pick listings for the jurisdiction they are assigned to 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/users/JurisdictionAndListingSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                          error={!!errors?.user_listings}
+                          errorMessage={t("errors.requiredFieldError")}
+                          validation={{ required: true }}
+                          dataTestId={`listings_${jurisdictionLabel}`}
+                        />
+                      </GridCell>
+                    </GridSection>
+                  </GridCell>
+                )
+              })}
```
Comment (seanmalbert) [inline sites/partners/src/users/JurisdictionAndListingSelection.tsx:151]: This looks pretty similar to what is below starting on line 161. Can these use the same component?

**Review by seanmalbert — APPROVED** [review]
Looks good, @YazeedLoonat . I'm wondering if you can clean up `sites/partners/src/users/JurisdictionAndListingSelection.tsx` a little more. See my comment on that file.

**Comment by YazeedLoonat** [comment]
Hey @seanmalbert after fidgeting with this, I think since what this is really doing is rendering an array of jsx elements it might be better to leave this as a function instead of making it a stand alone component. So currently in this pr its a function that produces the results

lemme know if you think its okay or if we really should make it a stand alone component

**Comment by seanmalbert** [comment]
@YazeedLoonat , thank you for the update. Can you please fix the failing backend test before merging.

**Pull request merged**